### ISSUE TITLE
Is it possible to set the code-input element to read-only?

### ISSUE THREAD

**Issue by vcdomith** [WebCoder49/code-input#59]
I was wondering if it is possible to set this element to a read-only state, so that in one page you write the code then it is shown in another page without the user being able to edit it.

**Comment by WebCoder49** [comment]
The `disabled` attribute currently can set this into a read-only state but there is a bug which means it cannot be reversed (as it reads the new value as undefined and sets the textarea's disabled attribute to "undefined" rather than removing it)!

I will try to fix the bug this weekend, or if you understand the codebase and want to do it via a pull request, please let me know.

**Comment by WebCoder49** [comment]
Here's the JavaScript code:

```javascript
let codeInput = document.querySelector("code-input"); // Or get your element in another way
codeInput.setAttribute("disabled"); // Make read-only - this currently works.
codeInput.removeAttribute("disabled"); // Make editable again - the bug with this needs to be fixed.
```

**Comment by vcdomith** [comment]
That's perfect, it'll work wonders, as the code-input editor is another template I don't think it'll interfere. Thank you very much! This element is a life-saver!

**Comment by WebCoder49** [comment]
I will try to implement the bug fix this weekend - please could I check that you are using the latest version 2.0?

**Comment by WebCoder49** [comment]
I will try to implement the bug fix this weekend - please could I check that you are using the latest version 2.0?

**Comment by WebCoder49** [comment]
I have released this functionality in v2.0.3. Please let me know if it doesn't work.

**Comment by WebCoder49** [comment]
For your information, please see the updated code snippets above (it turns out I was slightly wrong at first). If you want, you could use the [`readonly` attribute as mentioned here instead of `disabled`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#:~:text=readonly).
